src/browser/ui/ReactDOMComponent.js

```diff
@@ -232,6 +233,26 @@ function processChildContext(context, tagName) {
   return context;
 }
 
+var onlyNestTextTags = {
+  'option': true
+};
+
+function getChildrenTextContent(tag, children) {
+  var childrenContent = '';
+  ReactChildren.forEach(children, function (c) {
+    var childType = typeof c;
+    if (childType !== 'string' && childType !== 'number') {
+      // not null/undefined/boolean
+      if (c !== null) {
+        warning(false, '%s can not have complex children/content', tag);
```

Comment:
@spicyj Could it not make sense to `toString` them instead of skipping, making it even more prominent that some invalid data has been passed to something that only accepts text.